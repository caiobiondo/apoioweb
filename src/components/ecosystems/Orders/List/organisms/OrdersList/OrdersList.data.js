import gql from 'graphql-tag';

const ITEMS_PER_PAGE = 10;

export const OrdersListQuery = gql`
  query OrdersListQuery($offset: Int!, $limit: Int!) {
    orders(offset: $offset, limit: $limit) {
      codigoPedido
      dataPedido
      ciclo
      dataPrevisaoEntrega
      valor
      pontos
      valorLucro
      status
      statusTipo
    }
  }
`;

export const updateQuery = (previousResult, { fetchMoreResult }) => {
  if (!fetchMoreResult) {
    return previousResult;
  }
  return Object.assign({}, previousResult, {
    orders: [...previousResult.orders, ...fetchMoreResult.orders],
  });
};

export const OrdersListQueryOptions = {
  options(props) {
    return {
      variables: {
        limit: ITEMS_PER_PAGE,
        offset: 0,
      },
      forceFetch: true,
    };
  },
  props({ data }) {
    return {
      data,
      loading: data.loading,
      orders: data.orders,
      fetchMore() {
        return data.fetchMore({
          variables: {
            offset: data.orders.length,
          },
          updateQuery: updateQuery,
        });
      },
    };
  },
};
