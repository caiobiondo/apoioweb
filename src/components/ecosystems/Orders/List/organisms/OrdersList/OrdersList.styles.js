import styled from 'styled-components';
import { Responsive } from '@entria/components';
import { spSmall, spMedium, spPage } from 'styles/spacing';
import { LoadingWrapperStyle } from 'styles/mixins';

export const List = styled.ul`
  padding-left: 0;
  padding-top: ${spMedium};
  margin: ${spMedium};

  @media (max-width: ${Responsive.VIEWPORT.MEDIUM}px) {
    margin: ${spSmall};
  }
`;

export const LoadingWrapper = LoadingWrapperStyle;

export const scrolledContainer = {
  flex: '1 1 0',
  flexDirection: 'column',
  justifyContent: 'center',
  paddingBottom: spPage,
};

export const fullContainer = {
  height: '100%',
};
