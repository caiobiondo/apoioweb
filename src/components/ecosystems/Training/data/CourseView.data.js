import gql from 'graphql-tag';
import { Origem } from 'config';

export const CourseViewQuery = gql`
  query CourseViewQuery(
    $sellerId: Int!
    $courseId: Int!
    $ciclo: Int
    $setor: Int
    $gerenciaMercado: Int
    $grupo: Int
    $papelDaConsultora: Int
    $canal: Int
    $gerenciaDeVendas: Int
    $regiao: Int
    $origem: String!
  ) {
    course(
      sellerId: $sellerId
      courseId: $courseId
      ciclo: $ciclo
      setor: $setor
      gerenciaMercado: $gerenciaMercado
      grupo: $grupo
      papelDaConsultora: $papelDaConsultora
      canal: $canal
      gerenciaDeVendas: $gerenciaDeVendas
      regiao: $regiao
      origem: $origem
    ) {
      id
      accessToken
      categoryTitle
      clientIdentifier
      clientSecrets
      courseContent {
        video
        html5
        web
        videoEmbedUrl
        scorm
      }
      dateUpload
      durationInSeconds
      description
      generalRating
      isfavorite
      myRating
      ratedByYou
      relatedCourses {
        id
        title
        thumbnail
        type
        url
      }
      status
      stoppedAt
      thumbnail
      title
      type
      views
    }
  }
`;

export const CourseViewQueryOptions = {
  options(props) {
    return {
      variables: {
        sellerId: props.user.codigo,
        courseId: props.courseId,
        ciclo:
          props.user.estrutura.ciclo &&
          props.user.estrutura.ciclo[0] &&
          props.user.estrutura.ciclo[0].numero,
        setor: props.user.estrutura.codigoTipo > 3 ? props.user.estrutura.setor.codigo : 0,
        gerenciaMercado: props.user.estrutura.gerenciaMercado.codigo,
        grupo: props.user.estrutura.codigoTipo > 4 ? props.user.estrutura.codigo : 0,
        papelDaConsultora: props.user.cdPapelAtivo,
        canal: props.user.cdCanalCaptacao,
        gerenciaDeVendas:
          props.user.estrutura.codigoTipo > 2 ? props.user.estrutura.gerenciaVenda.codigo : 0,
        regiao:
          props.user.estrutura.codigoTipo > 1 ? props.user.estrutura.regiaoEstrategica.codigo : 0,
        origem: Origem,
      },
      fetchPolicy: 'cache-and-network',
      //fetchPolicy: 'no-cache',
    };
  },

  props({ data }) {
    return {
      data,
      loading: data.loading,
      course: data.course,
      refetch: data.refetch,
    };
  },
};
