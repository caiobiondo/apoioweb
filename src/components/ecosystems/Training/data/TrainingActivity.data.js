import gql from 'graphql-tag';
import { Origem } from 'config';

export const ActivityViewQuery = gql`
  query ActivityViewQuery(
    $sellerId: Int!
    $activityId: Int!
    $ciclo: Int
    $setor: Int
    $gerenciaMercado: Int
    $grupo: Int
    $papelDaConsultora: Int
    $canal: Int
    $gerenciaDeVendas: Int
    $regiao: Int
    $origem: String!
  ) {
    activity(
      sellerId: $sellerId
      activityId: $activityId
      ciclo: $ciclo
      setor: $setor
      gerenciaMercado: $gerenciaMercado
      grupo: $grupo
      papelDaConsultora: $papelDaConsultora
      canal: $canal
      gerenciaDeVendas: $gerenciaDeVendas
      regiao: $regiao
      origem: $origem
    ) {
      status
      message
      totalOfElement
      results {
        id
        finished
        name
        type
        accessToken
        courseContent {
          video
          html5
          web
          videoEmbedUrl
          html5Embed
          scorm
        }
        assessment {
          id
          title
          questions {
            id
            description
            type
            alternatives {
              id
              description
            }
          }
        }
      }
    }
  }
`;

export const ActivityViewQueryOptions = {
  options(props) {
    return {
      variables: {
        sellerId: props.user.codigo,
        activityId: props.activityId,
        ciclo:
          props.user.estrutura.ciclo &&
          props.user.estrutura.ciclo[0] &&
          props.user.estrutura.ciclo[0].numero,
        setor: props.user.estrutura.codigoTipo > 3 ? props.user.estrutura.setor.codigo : 0,
        gerenciaMercado: props.user.estrutura.gerenciaMercado.codigo,
        grupo: props.user.estrutura.codigoTipo > 4 ? props.user.estrutura.codigo : 0,
        papelDaConsultora: props.user.cdPapelAtivo,
        canal: props.user.cdCanalCaptacao,
        gerenciaDeVendas:
          props.user.estrutura.codigoTipo > 2 ? props.user.estrutura.gerenciaVenda.codigo : 0,
        regiao:
          props.user.estrutura.codigoTipo > 1 ? props.user.estrutura.regiaoEstrategica.codigo : 0,
        origem: Origem,
      },
      fetchPolicy: 'network-only',
    };
  },

  props({ data }) {
    return {
      data,
      loading: data.loading,
      activity: data.activity,
      refetch: data.refetch,
    };
  },
};
