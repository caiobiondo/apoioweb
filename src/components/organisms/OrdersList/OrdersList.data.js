import gql from 'graphql-tag';

const OrdersListQuery = gql`
  query OrdersListQuery {
    orders(id: 1) {
      cycle
      date
      estimatedDeliveryDate
      orderNumber
      orderValue
      status
      totalScore
    }
  }
`;

export default OrdersListQuery;
