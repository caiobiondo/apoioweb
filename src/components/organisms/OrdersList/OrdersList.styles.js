import styled from 'styled-components';

export const List = styled.ul`
  padding-left: 0;
`;

export const LoadingWrapper = styled.div`
  text-align: center;
`;
