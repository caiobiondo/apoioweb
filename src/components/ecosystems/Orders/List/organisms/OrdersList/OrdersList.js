import React, { Component } from 'react';
import { Loading, CircularProgress, Paper } from 'natura-ui';
import { List, LoadingWrapper, scrolledContainer } from './OrdersList.styles';
import { OrdersListQuery, OrdersListQueryOptions } from './OrdersList.data';
import Order from '../../molecules/Order';
import EmptyList from 'components/molecules/EmptyList/EmptyList';
import { graphql } from 'react-apollo';
import { injectIntl } from 'react-intl';
import { formatDate, formatCurrency } from 'locale/utils';
import { CAPTA_ORDER_LIST_URL } from 'config';

import InfiniteScroll from 'components/organisms/InfiniteScroll';

const renderOrder = (order, importing, intl) => {
  const orderDate = formatDate(order.dataPedido, intl, '-');
  const orderEstimatedDeliveryDate = formatDate(order.dataPrevisaoEntrega, intl, '-');
  const orderValue = formatCurrency(order.valor, intl, '-');

  // Order details feature being hidden because capta's api is blocking it due to single sign on
  // let url = `${ROUTE_PREFIX}/my-orders/detail/${order.codigoPedido}`;
  // if (importing) {
  //   url = `${ROUTE_PREFIX}/my-stock/import/orders/detail/${order.codigoPedido}`;
  // }
  const url = CAPTA_ORDER_LIST_URL;

  return (
    <Order
      importing={importing}
      key={order.codigoPedido}
      statusType={order.statusTipo}
      left={{
        body: {
          orderDate: orderDate,
          orderEstimatedDeliveryDate: orderEstimatedDeliveryDate,
        },
        header: {
          orderNumber: `#${order.codigoPedido}`,
        },
      }}
      middle={{
        body: {
          orderTotalScore: order.pontos,
          orderCycle: order.ciclo,
        },
        header: {
          orderValue: orderValue,
        },
      }}
      right={{
        details: url,
        status: order.status,
      }}
    />
  );
};

export class OrdersList extends Component {
  componentWillReceiveProps({ loading, orders, onLoadFinished }) {
    if (!loading && onLoadFinished) {
      this.props.onLoadFinished(this._empty(loading, orders), this._loading(loading, orders));
    }
  }

  _loading(loading, orders) {
    return loading && !orders;
  }

  _empty(loading, orders) {
    return !loading && (!orders || orders.length === 0);
  }

  render() {
    const { loading, orders, fetchMore, importing, intl, hasNextPage } = this.props;

    if (this._loading(loading, orders)) {
      return <Loading background="transparent" />;
    }

    return (
      <Paper style={scrolledContainer}>
        <InfiniteScroll
          onScroll={fetchMore}
          hasMore={hasNextPage}
          loading={loading}
          debounce={500}
          items={orders}
          emptyList={
            <EmptyList
              icon="ico_box"
              titleId="ordersEmptyList"
              descriptionId="ordersWithoutOrders"
            />
          }
          loader={
            <LoadingWrapper>
              <CircularProgress thickness={2} />
            </LoadingWrapper>
          }
        >
          <List>{orders && orders.map(order => renderOrder(order, importing, intl))}</List>
        </InfiniteScroll>
      </Paper>
    );
  }
}

export const OrdersListWithIntl = injectIntl(OrdersList);

export default graphql(OrdersListQuery, OrdersListQueryOptions)(OrdersListWithIntl);
