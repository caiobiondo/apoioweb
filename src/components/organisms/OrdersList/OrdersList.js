import React, { Component } from 'react';
import { Loading, CircularProgress, Paper } from 'natura-ui';
import { List, LoadingWrapper, scrolledContainer, fullContainer } from './OrdersList.styles';
import { OrdersListQuery, OrdersListQueryOptions } from './OrdersList.data';
import Order from 'components/molecules/Order/Order';
import EmptyOrders from 'components/molecules/EmptyOrders/EmptyOrders';
import { graphql } from 'react-apollo';
import { injectIntl } from 'react-intl';
import { formatDate, formatCurrency } from 'locale/utils';
import InfiniteScroll from 'react-infinite-scroller';

const renderOrder = (order, intl) => {
  const orderDate = formatDate(order.dataPedido, intl, '-');
  const orderEstimatedDeliveryDate = formatDate(order.dataPrevisaoEntrega, intl, '-');
  const orderValue = formatCurrency(order.valor, intl, '-');
  const orderProfitsValue = formatCurrency(order.valorLucro, intl, '-');
  return (
    <Order
      key={order.codigoPedido}
      statusType={order.statusTipo}
      left={{
        body: {
          orderDate: orderDate,
          orderCycle: order.ciclo,
          orderEstimatedDeliveryDate: orderEstimatedDeliveryDate,
        },
        header: {
          orderNumber: `#${order.codigoPedido}`,
        },
      }}
      middle={{
        body: {
          orderTotalScore: order.pontos,
          orderProfitsValue: orderProfitsValue,
        },
        header: {
          orderValue: orderValue,
        },
      }}
      right={{
        details: `my-orders/${order.codigoPedido}`,
        status: order.status,
      }}
    />
  );
};

export class OrdersList extends Component {
  componentWillReceiveProps({ loading, orders, onLoadFinished }) {
    if (!loading && onLoadFinished) {
      this.props.onLoadFinished(this._empty(loading, orders), this._loading(loading, orders));
    }
  }

  _loading(loading, orders) {
    return loading && !orders;
  }

  _empty(loading, orders) {
    return !loading && (!orders || orders.length === 0);
  }

  render() {
    const { loading, orders, fetchMore, intl } = this.props;
    if (this._loading(loading, orders)) {
      return <Loading background="transparent" />;
    }
    if (this._empty(loading, orders)) {
      return (
        <Paper style={fullContainer}>
          <EmptyOrders />
        </Paper>
      );
    }
    return (
      <Paper style={scrolledContainer}>
        <InfiniteScroll
          loadMore={fetchMore}
          hasMore={false}
          loader={
            <LoadingWrapper>
              <CircularProgress thickness={2} />
            </LoadingWrapper>
          }
        >
          <List>{orders.map(order => renderOrder(order, intl))}</List>
        </InfiniteScroll>
      </Paper>
    );
  }
}

export const OrdersListWithIntl = injectIntl(OrdersList);

export default graphql(OrdersListQuery, OrdersListQueryOptions)(OrdersListWithIntl);
