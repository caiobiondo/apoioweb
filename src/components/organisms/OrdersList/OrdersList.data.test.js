import { OrdersListQuery } from './OrdersList.data';

describe('OrdersListQuery', () => {
  it('should be the correct query', () => {
    expect(OrdersListQuery).toMatchSnapshot();
  });
});
