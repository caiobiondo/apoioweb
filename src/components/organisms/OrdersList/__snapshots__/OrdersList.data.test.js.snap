// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrdersListQuery should be the correct query 1`] = `
Object {
  "definitions": Array [
    Object {
      "directives": Array [],
      "kind": "OperationDefinition",
      "name": Object {
        "kind": "Name",
        "value": "OrdersListQuery",
      },
      "operation": "query",
      "selectionSet": Object {
        "kind": "SelectionSet",
        "selections": Array [
          Object {
            "alias": null,
            "arguments": Array [
              Object {
                "kind": "Argument",
                "name": Object {
                  "kind": "Name",
                  "value": "offset",
                },
                "value": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "offset",
                  },
                },
              },
              Object {
                "kind": "Argument",
                "name": Object {
                  "kind": "Name",
                  "value": "limit",
                },
                "value": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "limit",
                  },
                },
              },
            ],
            "directives": Array [],
            "kind": "Field",
            "name": Object {
              "kind": "Name",
              "value": "orders",
            },
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": null,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "codigoPedido",
                  },
                  "selectionSet": null,
                },
                Object {
                  "alias": null,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "dataPedido",
                  },
                  "selectionSet": null,
                },
                Object {
                  "alias": null,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "ciclo",
                  },
                  "selectionSet": null,
                },
                Object {
                  "alias": null,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "dataPrevisaoEntrega",
                  },
                  "selectionSet": null,
                },
                Object {
                  "alias": null,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "valor",
                  },
                  "selectionSet": null,
                },
                Object {
                  "alias": null,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "pontos",
                  },
                  "selectionSet": null,
                },
                Object {
                  "alias": null,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "valorLucro",
                  },
                  "selectionSet": null,
                },
                Object {
                  "alias": null,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "status",
                  },
                  "selectionSet": null,
                },
                Object {
                  "alias": null,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "statusTipo",
                  },
                  "selectionSet": null,
                },
              ],
            },
          },
        ],
      },
      "variableDefinitions": Array [
        Object {
          "defaultValue": null,
          "kind": "VariableDefinition",
          "type": Object {
            "kind": "NonNullType",
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "Int",
              },
            },
          },
          "variable": Object {
            "kind": "Variable",
            "name": Object {
              "kind": "Name",
              "value": "offset",
            },
          },
        },
        Object {
          "defaultValue": null,
          "kind": "VariableDefinition",
          "type": Object {
            "kind": "NonNullType",
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "Int",
              },
            },
          },
          "variable": Object {
            "kind": "Variable",
            "name": Object {
              "kind": "Name",
              "value": "limit",
            },
          },
        },
      ],
    },
  ],
  "kind": "Document",
  "loc": Object {
    "end": 257,
    "start": 0,
  },
}
`;

exports[`OrdersListQuery should be the correct query props 1`] = `
Object {
  "error": undefined,
  "fetchMore": [Function],
  "loading": true,
  "orders": Array [],
}
`;
